java/src/main/java/com/logicalclocks/hsfs/engine/StreamFeatureGroupEngine.java

```diff
@@ -131,53 +203,15 @@ public <S> StreamFeatureGroup save(StreamFeatureGroup featureGroup, S dataset, L
       featureGroup.setFeatures(features);
     }
 
-    // Write the online dataframe
-    insert(featureGroup, utils.sanitizeFeatureNames(dataset), HudiOperationType.BULK_INSERT,
-        SaveMode.APPEND, writeOptions);
-
-    return featureGroup;
-  }
-
-  @SneakyThrows
-  public <S> Object insertStream(StreamFeatureGroup streamFeatureGroup, S featureData, String queryName,
-                                 String outputMode, boolean awaitTermination, Long timeout,  String checkpointLocation,
-                                 Map<String, String> writeOptions) {
-
-    if (streamFeatureGroup.getValidationType() != ValidationType.NONE) {
-      LOGGER.info("ValidationWarning: Stream ingestion for feature group `" + streamFeatureGroup.getName()
-                    + "`, with version `" + streamFeatureGroup.getVersion() + "` will not perform validation.");
-    }
-
-    if (writeOptions == null) {
-      writeOptions = new HashMap<>();
-    }
-
-    return SparkEngine.getInstance().writeStreamDataframe(streamFeatureGroup,
-      utils.sanitizeFeatureNames(featureData), queryName, outputMode, awaitTermination, timeout, checkpointLocation,
-      utils.getKafkaConfig(streamFeatureGroup, writeOptions));
-  }
-
-  public <S> void insert(StreamFeatureGroup streamFeatureGroup, S featureData, HudiOperationType operation,
-                         SaveMode saveMode, Map<String, String> writeOptions) throws FeatureStoreException, IOException,
-      ParseException {
-
-    Integer validationId = null;
-    if (streamFeatureGroup.getValidationType() != ValidationType.NONE) {
-      FeatureGroupValidation validation = streamFeatureGroup.validate(featureData, true);
-      if (validation != null) {
-        validationId = validation.getValidationId();
-      }
-    }
-
-    if (saveMode == SaveMode.OVERWRITE) {
-      // If we set overwrite, then the directory will be removed and with it all the metadata
-      // related to the feature group will be lost. We need to keep them.
-      // So we call Hopsworks to manage to truncate the table and re-create the metadata
-      // After that it's going to be just a normal append
-      featureGroupApi.deleteContent(streamFeatureGroup);
+    if (saveEmpty) {
```

Comment:
It's not exactly clear to me why sometimes we need to call this `saveEmpty` and sometimes we don't. I'm not sure what I am missing.

Reply:
good point. Its not really relevant here.